app/controllers/shipit/api/rollbacks_controller.rb

```diff
@@ -0,0 +1,24 @@
+
+module Shipit
+  module Api
+    class RollbacksController < BaseController
+      require_permission :deploy, :stack
+
+      params do
+        requires :sha, String, length: { in: 6..40 }
+        accepts :force, Boolean, default: false
+        accepts :env, Hash, default: {}
+      end
+      def create
+        commit = stack.commits.by_sha(params.sha) || param_error!(:sha, 'Unknown revision')
```

Comment:
It'd be a good idea to surface any errors from this API on ShipitNext - or perhaps, to the user's slack - when the button is pressed.